Source/SwiftLintBuiltInRules/Rules/Idiomatic/RedundantTypeAnnotationRule.swift

```diff
@@ -72,98 +124,183 @@ struct RedundantTypeAnnotationRule: OptInRule, SubstitutionCorrectableRule {
                 let myVar = Int(5)
               }
             }
-            """)
+            """),
+            Example("var num: Int = Int.random(0..<10)"): Example("var num = Int.random(0..<10)")
         ]
     )
 
-    func validate(file: SwiftLintFile) -> [StyleViolation] {
-        return violationRanges(in: file).map { range in
-            StyleViolation(
-                ruleDescription: Self.description,
-                severity: configuration.severity,
-                location: Location(file: file, characterOffset: range.location)
-            )
-        }
+    func makeRewriter(file: SwiftLintFile) -> (some ViolationsSyntaxRewriter)? {
+        Rewriter(
+            configuration: configuration,
+            locationConverter: file.locationConverter,
+            disabledRegions: disabledRegions(file: file)
+        )
     }
+}
 
-    func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
-        return (violationRange, "")
-    }
+private extension RedundantTypeAnnotationRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+            guard node.doesNotContainIgnoredAttributes(for: configuration) else {
```

Comment:
This is a good idea. However, it skips all other redundant type annotations nested in the variable declaration, e.g.

```swift
@IgnoreMe
var j: Int {
    let i: Int = Int(1) // will not trigger, but should
    return i
}
```

Therefore it's better to check the parent's parent of `PatternBindingSyntax` for the attribute.

Reply:
Good call! I've made the update as you suggested